tests/scripts/curves.pl

```diff
@@ -21,21 +21,25 @@
 #
 # Purpose
 #
-# To test the code dependencies on individual curves in each test suite. This
-# is a verification step to ensure we don't ship test suites that do not work
-# for some build options.
+# The purpose of this test script is to validate that the library works
+# with any combination of elliptic curves. To this effect, build the library
+# and run the test suite with each tested combination of elliptic curves.
 #
-# The process is:
-#       for each possible curve
-#           build the library and test suites with the curve disabled
-#           execute the test suites
-#
-# And any test suite with the wrong dependencies will fail.
+# Testing all 2^n combinations would be too much, so we only test 2*n:
 #
+# 1. Test with a single curve, for each curve. This validates that the
+#    library works with any curve, and in particular that curve-specific
+#    code is guarded by the proper preprocessor conditionals.
+# 2. Test with all curves except one, for each curve. This validates that
+#    the test cases have correct dependencies. Testing with a single curve
+#    doesn't validate this for tests that require more than one curve.
+
 # Usage: tests/scripts/curves.pl
 #
 # This script should be executed from the root of the project directory.
 #
+# Only curves that are enabled in config.h will be tested.
```

Comment:
"are tested" instead of "will be tested"?

Reply:
Why? “Will be tested”, as in that's what will happen later in the script.

Reply:
I understand "will be tested" is fine. It is just that the future there sounded a bit weird to me, probably because before the comments are descriptive and in present/imperative.